WeatherMonitoringSystem/Domain/Config/ConfigurationManager.cs

```diff
@@ -0,0 +1,56 @@
+﻿
+using WeatherMonitoringSystem.Config;
+
+namespace WeatherMonitoringSystem.Domain.Config
+{
+    using Newtonsoft.Json;
+    public class ConfigurationManager
+    {
+        private static ConfigurationManager _instance;
+        private static readonly object _lock = new object();
+
+        public BotConfig RainBot { get; private set; }
+        public BotConfig SunBot { get; private set; }
+        public BotConfig SnowBot { get; private set; }
+
+        private ConfigurationManager()
+        {
+                LoadConfiguration();
+        }
+
+        public static ConfigurationManager Instance
+        {
+            get
+            {
+                lock (_lock)
```

Comment:
why did you have to use locks here? I don't see any type of multi-threaded contexts.